src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -483,17 +482,20 @@ public FSDataInputStream getObject(String hostName, Path path) throws IOExceptio
 
     LOG.debug("List container for {} container {}", obj, container);
     ArrayList<FileStatus> tmpResult = new ArrayList<FileStatus>();
-    PaginationMap paginationMap = cObj.getPaginationMap(obj, pageListSize);
-    FileStatus fs = null;
     StoredObject previousElement = null;
-    for (Integer page = 0; page < paginationMap.getNumberOfPages(); page++) {
-      Collection<StoredObject> res = cObj.list(paginationMap, page);
-      if (page == 0 && (res == null || res.isEmpty())) {
+    boolean moreData = true;
+    String marker = null;
+    FileStatus fs = null;
+    while (moreData) {
+      Collection<StoredObject> res = cObj.list(obj, marker, pageListSize);
+      moreData = (res.size() == pageListSize);
+      if (marker == null && (res == null || res.isEmpty() || res.size() == 0)) {
         FileStatus[] emptyRes = {};
         LOG.debug("List {} in container {} is empty", obj, container);
         return emptyRes;
       }
       for (StoredObject tmp : res) {
+        marker = tmp.getAsObject().getName();
```

Comment:
Good suggestion Gil.  

The while loop spins while there is moreData.  Each time there is moreData it reads from the list using the prior marker.  We therefore need to set the marker each time we go through the while loop for the next read.  The marker is just the name of the last object that was returned.  We can pull it out of the for loop and do something exactly like you suggested.  I can do that, test it, and get back to you.